lib/mgmt_msg_native.h

```diff
@@ -231,6 +233,23 @@ _Static_assert(sizeof(struct mgmt_msg_tree_data) ==
 		       offsetof(struct mgmt_msg_tree_data, result),
 	       "Size mismatch");
 
+/**
+ * struct mgmt_msg_get_data - frontend get-data request.
+ *
+ * @result_type: ``LYD_FORMAT`` for the returned result.
+ * @xpath: the query for the data to return.
+ */
+struct mgmt_msg_get_data {
+	struct mgmt_msg_header;
+	uint8_t result_type;
+	uint8_t resv2[7];
+
+	alignas(8) char xpath[];
+};
```

Comment:
Why not just replace `mgmt_msg_get_tree` with this? Even with the expanded functionality coming in later commits? I don't see any point in having a specialized `get-tree` message that can be exactly replaced by this one with a couple extra params.

Reply:
I think the code is more clear if we have separate messages for the frontend and for the backend. There are new fields in the frontend message and there will be more later (like `datastore`, `with-defaults`, etc., to cover the whole NETCONF/RESTCONF functionality). And all these new fields don't make sense on the backend.